ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -745,13 +748,30 @@ object BigDiffy extends Command with Serializable {
     val result = inputMode match {
       case "avro" =>
         if (rowRestriction.isDefined) {
-          throw new IllegalArgumentException(s"rowRestriction cannot be passed for avro inputs")
+          throw new IllegalArgumentException("rowRestriction cannot be passed for avro inputs")
+        }
+
+        val lhsSchema = avroFileSchema(lhs, sc.options)
+        val rhsSchema = avroFileSchema(rhs, sc.options)
+
+        val lhsReader = SchemaCompatibility.checkReaderWriterCompatibility(lhsSchema, rhsSchema)
+        val rhsReader = SchemaCompatibility.checkReaderWriterCompatibility(rhsSchema, lhsSchema)
+
+        import SchemaCompatibility.SchemaCompatibilityType._
+        val schema = (lhsReader.getType, rhsReader.getType) match {
+          case (COMPATIBLE, COMPATIBLE) =>
+            if (lhsSchema != rhsSchema) {
+              logger.warn("Avro schemas are compatible, but not equal. Using schema from {}", rhs)
+            }
+            rhsSchema
+          case (COMPATIBLE, INCOMPATIBLE) =>
```

Comment:
This is a change in underlying functionality, IMO We should also warn in these cases rather than proceed transparently

Reply:
Honestly I still think we enforce RHS backwards compatibility unless otherwise shown to be necessary, but if we are changing functionality/flexibility then we need to do so in a way that is transparent to users.

I'll leave the actual decision here down to current members of the owning team

Reply:
reverted to previous behavior using the `SchemaValidatorBuilder` that thows a `SchemaValidationException` with detailed error in case of incompatibility.